Jenkinsfile

```diff
@@ -17,6 +17,17 @@ pipeline {
         junit 'features/reports/*.xml'
       }
     }
+  }
 
+  post {
+    always {
+      sh 'docker run -i --rm -v $PWD:/src -w /src alpine/git clean -fxd'
```

Comment:
Yeah we're close to a solution, but this is the best we have so far, from @apotterri. We've been talking about this in #about-jenkins if you'd like to join.